File: src/models/note.js
//! STEPS
//! 1.DEFINE SCHEMA => Model details in schema => Note: id, userId, title, content, dateAdded => These fields are called schema
//! 2. CREATE MODEL => <Model name> <Schema> Note

const mongoose = require("mongoose");

//? Schema
const noteSchema = mongoose.Schema({
  id: {
    type: String,
    unique: true,
    required: true,
  },
  userId: {
    type: String,
    required: true,
  },
  title: {
    type: String,
    required: true,
  },
  content: {
    type: String
  },
  dateAdded: {
    type: Date,
    default: Date.now
  },
});

//? Model
module.exports = mongoose.model("Note", noteSchema)

File: lib/models/note.dart
class Note {
  String? id;
  String? userId;
  String? title;
  String? content;
  DateTime? dateAdded;

  //! Default constructor
  Note({
    this.id,
    this.userId,
    this.title,
    this.content,
    this.dateAdded,
  });

  //! Object
  factory Note.fromMap(Map<String, dynamic> map) => Note(
        id: map["id"],
        userId: map["userId"],
        title: map["title"],
        content: map["content"],
        dateAdded: DateTime.tryParse(
          map["dateAdded"],
        ),
      );

  Map<String, dynamic> toMap() => {
        "id": id,
        "userId": userId,
        "title": title,
        "content": content,
        "dateAdded": dateAdded!.toIso8601String(),
      };
}
